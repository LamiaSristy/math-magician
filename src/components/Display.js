import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.module.css';

/* eslint-disable */
const Display = ({ total, next, operation }) => {
  return (
    <div className={styles.display}>
      <p className="mb-0 text-right w-100">{next || total}</p>
      <span className={styles.displaySubtitle}>
        {next && (total || null)} {operation || null}
      </span>
    </div>
  );
};

Display.defaultProps = {
  total: '0',
  next: undefined,
  operation: undefined,
};

Display.propTypes = {
  total: PropTypes.string,
  operation: PropTypes.string,
  next: PropTypes.string,
};

export default Display;
