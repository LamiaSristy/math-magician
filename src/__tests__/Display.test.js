import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

describe('Display', () => {
  test('Render the display component  correctly', () => {
    const displayTree = renderer.create(<Display />).toJSON();
    expect(displayTree).toMatchSnapshot();
  });
});
