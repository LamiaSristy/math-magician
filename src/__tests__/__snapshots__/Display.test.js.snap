// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Render the display component  correctly 1`] = `
<div
  className="display"
>
  <p
    className="mb-0 text-right w-100"
  >
    0
  </p>
  <span
    className="displaySubtitle"
  >
     
  </span>
</div>
`;
